manifests/init.pp

```diff
@@ -12,41 +12,84 @@
 # @param shipper [Hash] Will be converted to YAML to create the optional shipper section of the winlogbeat config
 # @param logging [Hash] Will be converted to YAML to create the optional logging section of the winlogbeat config
+# @param conf_template [String] The configuration template to use to generate the main winlogbeat.yml config file
+# @param download_url [String] The URL of the zip file that should be downloaded to install winlogbeat
+# @param install_dir [String] Where winlogbeat should be installed
 # @param tmp_dir [String] Where winlogbeat should be temporarily downloaded to so it can be installed
 # @param event_logs [Hash] Event_logs that will be forwarded.
 # @param event_logs_merge [Boolean] Whether $event_logs should merge all hiera sources, or use simple automatic parameter lookup
 
 class winlogbeat (
-  $conf_template    = $winlogbeat::params::conf_template,
-  $config_file      = $winlogbeat::params::config_file,
-  $download_url     = $winlogbeat::params::download_url,
-  $install_dir      = $winlogbeat::params::install_dir,
-  $outputs          = $winlogbeat::params::outputs,
-  $package_ensure   = $winlogbeat::params::package_ensure,
-  $registry_file    = $winlogbeat::params::registry_file,
-  $service_enable   = $winlogbeat::params::service_enable,
-  $service_ensure   = $winlogbeat::params::service_ensure,
-  $shipper          = $winlogbeat::params::shipper,
-  $logging          = $winlogbeat::params::logging,
-  $tmp_dir          = $winlogbeat::params::tmp_dir,
-  $event_logs       = {},
-  $event_logs_merge = false,
+  $major_version        = undef,
+  $package_ensure       = $winlogbeat::params::package_ensure,
+  $service_ensure       = $winlogbeat::params::service_ensure,
+  $service_enable       = $winlogbeat::params::service_enable,
+  $service_provider     = $winlogbeat::params::service_provider,
+  $registry_file        = $winlogbeat::params::registry_file,
+  $config_file          = $winlogbeat::params::config_file,
+  $outputs              = $winlogbeat::params::outputs,
+  $shipper              = $winlogbeat::params::shipper,
+  $logging              = $winlogbeat::params::logging,
+  $run_options          = $winlogbeat::params::run_options,
+  $conf_template        = undef,
+  $download_url         = undef,
+  $install_dir          = $winlogbeat::params::install_dir,
+  $tmp_dir              = $winlogbeat::params::tmp_dir,
+  #### v5 only ####
+  $use_generic_template = $winlogbeat::params::use_generic_template,
+  $beat_name            = $winlogbeat::params::beat_name,
+  $tags                 = $winlogbeat::params::tags,
+  $queue_size           = $winlogbeat::params::queue_size,
+  $max_procs            = $winlogbeat::params::max_procs,
+  $fields               = $winlogbeat::params::fields,
+  $fields_under_root    = $winlogbeat::params::fields_under_root,
+  $metrics              = undef,
+  #### End v5 onlly ####
```

Comment:
Copied from here: https://github.com/pcfens/puppet-filebeat/blob/master/manifests/init.pp#L86

I'll need to fix it there also...